sympy/logic/boolalg.py

```
@@ -412,7 +412,19 @@ class BooleanFunction(Application, Boolean):
     is_Boolean = True
 
     def _eval_simplify(self, ratio, measure, rational, inverse):
-        return simplify_logic(self)
+        try:
+            args = [x._eval_simplify(
+                ratio=ratio,
+                measure=measure,
+                rational=rational,
+                inverse=inverse) for x in self.args]
+            rv = self.func(*args)
+        except:
```

Don't use bare except